modules/magniteAnalyticsAdapter.js

```diff
@@ -0,0 +1,820 @@
+const MODULE_NAME = 'Magnite Analytics';
+const BID_REJECTED_IPF = 'rejected-ipf';
+
+// List of known rubicon aliases
+// This gets updated on auction init to account for any custom aliases present
+let rubiconAliases = ['rubicon'];
+
+const pbsErrorMap = {
+  1: 'timeout-error',
+  2: 'input-error',
+  3: 'connect-error',
+  4: 'request-error',
+  999: 'generic-error'
+}
+
+let prebidGlobal = getGlobal();
+const {
+  EVENTS: {
+    AUCTION_INIT,
+    AUCTION_END,
+    BID_REQUESTED,
+    BID_RESPONSE,
+    BIDDER_DONE,
+    BID_TIMEOUT,
+    BID_WON,
+    BILLABLE_EVENT
+  },
+  STATUS: {
+    GOOD,
+    NO_BID
+  },
+  BID_STATUS: {
+    BID_REJECTED
+  }
+} = CONSTANTS;
+
+// The saved state of rubicon specific setConfig controls
+export let rubiConf;
+// Saving state of all our data we want
+let cache;
+const resetConfs = () => {
+  cache = {
+    auctions: {},
+    auctionOrder: [],
+    timeouts: {},
+    billing: {},
+    pendingEvents: {},
+    eventPending: false
+  }
+  rubiConf = {
+    pvid: generateUUID().slice(0, 8),
+    analyticsEventDelay: 0,
+    analyticsBatchTimeout: 5000,
+    dmBilling: {
+      enabled: false,
+      vendors: [],
+      waitForAuction: true
+    }
+  }
+}
+resetConfs();
+
+config.getConfig('rubicon', config => {
+  mergeDeep(rubiConf, config.rubicon);
+  if (deepAccess(config, 'rubicon.updatePageView') === true) {
+    rubiConf.pvid = generateUUID().slice(0, 8)
+  }
+});
+
+// pbs confs
+let serverConfig;
+config.getConfig('s2sConfig', ({ s2sConfig }) => {
+  serverConfig = s2sConfig;
+});
+
+const DEFAULT_INTEGRATION = 'pbjs';
+
+const adUnitIsOnlyInstream = adUnit => {
+  return adUnit.mediaTypes && Object.keys(adUnit.mediaTypes).length === 1 && deepAccess(adUnit, 'mediaTypes.video.context') === 'instream';
+}
+
+const sendPendingEvents = () => {
+  cache.pendingEvents.trigger = 'batchedEvents';
+  sendEvent(cache.pendingEvents);
+  cache.pendingEvents = {};
+  cache.eventPending = false;
+}
+
+const addEventToQueue = (event, auctionId, eventName) => {
+  // If it's auction has not left yet, add it there
+  if (cache.auctions[auctionId] && !cache.auctions[auctionId].sent) {
+    cache.auctions[auctionId].pendingEvents = mergeDeep(cache.auctions[auctionId].pendingEvents, event);
+  } else if (rubiConf.analyticsEventDelay > 0) {
+    // else if we are trying to batch stuff up, add it to pending events to be fired
+    cache.pendingEvents = mergeDeep(cache.pendingEvents, event);
+
+    // If no event is pending yet, start a timer for them to be sent and attempted to be gathered together
+    if (!cache.eventPending) {
+      setTimeout(sendPendingEvents, rubiConf.analyticsEventDelay);
+      cache.eventPending = true;
+    }
+  } else {
+    // else - send it solo
+    event.trigger = `solo-${eventName}`;
+    sendEvent(event);
+  }
+}
+
+const sendEvent = payload => {
+  const event = {
+    ...getTopLevelDetails(),
+    ...payload
+  }
+  ajax(
+    rubiConf.analyticsEndpoint || endpoint,
```

Comment:
It is honestly so the QA team can change the endpoint dynamically in their tests. Since DMS Build auto-adds the endpoint. But since then we have updated it to be controlled by puppet so we can remove this! Thanks.